src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -187,6 +187,44 @@ public object GetMetric(string metric)
             return _configStore.GetEngineMetric(metric);
         }
 
+        public int Jump(string filename, int line)
+        {
+            try
+            {
+                _debuggedProcess.WorkerThread.RunOperation(() => _debuggedProcess.Jump(filename, line));
+            }
+            catch (InvalidCoreDumpOperationException)
+            {
+                return AD7_HRESULT.E_CRASHDUMP_UNSUPPORTED;
+            }
+            catch (Exception e)
+            {
+                _engineCallback.OnError(EngineUtils.GetExceptionDescription(e));
+                return Constants.E_ABORT;
+            }
+
+            return Constants.S_OK;
+        }
+
```

Comment:
Where do we use this overload?

Reply:
First I used this one, then added addresses as they are unique.
But we need to find a way to actually get all the addresses for a given source line from gdb.

Reply:
Did you find that `-symbol-list-lines` doesn't work correctly for this case? (See (LinesForFile)[https://github.com/microsoft/MIEngine/blob/29c6569861b2042c43f04d8d78b2566db2e2ef1f/src/MIDebugEngine/Engine.Impl/SourceLine.cs#L93]).

Regardless, I don't think you want these overloads since we always convert the source file + line to instruction before we get this far.

Reply:
This is what is used:
https://github.com/microsoft/MIEngine/blob/b2f1555b2f92e7e6a6090832a2145ac1658bc89a/src/MICore/CommandFactories/gdb.cs#L158-L160

I tested it with a template and only got 1 instantiation.

Reply:
Sorry if this is obvious: but did you double check that the two templates didn't get folded together by the compiler/linker.

Assuming that isn't it -- you could certainly try `-symbol-list-lines` to see if that does any better. If not, I am guessing we would need GDB support to make this feature work quite correctly in the template case.

Reply:
Didn't check but didn't use any optimizations either.

Reply:
```cpp
#include <stdexcept>

template <typename T>
int foo(T t)
{
	return 0;
}

int main(int argc, const char* argv[])
{
	return foo(0.5f) + foo(5);
}
```
```
`-symbol-list-lines test.cpp
lines: [{pc=0x0000555555555129,line=10},{pc=0x000055555555513d,line=11},{pc=0x000055555555514c,line=11},{pc=0x0000555555555156,line=11},{pc=0x0000555555555158,line=12},{pc=0x000055555555515f,line=0},{pc=0x000055555555515f,line=4},{pc=0x000055555555516c,line=6},{pc=0x0000555555555171,line=7},{pc=0x0000555555555173,line=0},{pc=0x0000555555555173,line=4},{pc=0x000055555555517e,line=6},{pc=0x0000555555555183,line=7},{pc=0x0000555555555185,line=0}]

`i li test.cpp:6
Line 6 of "test.cpp" starts at address 0x55555555516c <foo<float>(float)+13> and ends at 0x555555555171 <foo<float>(float)+18>.
```
list-lines does cover everything, even single instructions per line.
But it would need to be cached.

Reply:
I am thinking we could detect when to ignore repeated records based on if the line number is the same between records. For example, in your first example output, save the first entry (line 10), and the second entry (line 11) but then ignore the next two since those are also line 11.

I would agree that `-symbol-list-lines` could be quite verbose, but, unless I am missing something, it seems to be what we use today for EnumCodeContexts.

Reply:
> unless I am missing something, it seems to be what we use today for EnumCodeContexts.

Yes but it falls back to StartAddressesForLine in the end.

Reply:
> you could certainly try `-symbol-list-lines` to see if that does any better. If not, I am guessing we would need GDB support to make this feature work quite correctly in the template case.

https://sourceware.org/bugzilla/show_bug.cgi?id=26594